moveit_planners/trajopt/trajopt/CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(trajopt)
+
+add_compile_options(-std=c++11 -Wall -Wextra)
+
+find_package(catkin REQUIRED COMPONENTS
+  trajopt_sco
+  trajopt_utils
+  roscpp
+)
+
+find_package(Eigen3 REQUIRED)
+find_package(Boost COMPONENTS system python thread program_options REQUIRED)
+
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(JSONCPP jsoncpp)
+
+link_directories(
+    /usr/local/lib
+    /usr/local/lib64
+)
+
+set(TRAJOPT_SOURCE_FILES
+    src/trajectory_costs.cpp
+    src/kinematic_terms.cpp
+    src/problem_description.cpp
+    src/utils.cpp
+)
+
+catkin_package(
+  INCLUDE_DIRS
+    include
+    ${EIGEN3_INCLUDE_DIRS}
+  LIBRARIES
+    ${PROJECT_NAME}
+  CATKIN_DEPENDS
+    trajopt_sco
+    trajopt_utils
+    roscpp
+  DEPENDS
+    EIGEN3
+    Boost
+)
+
+include_directories(
+  include
+  ${catkin_INCLUDE_DIRS}
+  SYSTEM ${EIGEN3_INCLUDE_DIRS}
+  SYSTEM ${Boost_INCLUDE_DIRS}
+  SYSTEM ${JSONCPP_INCLUDE_DIRS}
+)
+
+add_library(${PROJECT_NAME} ${TRAJOPT_SOURCE_FILES})
+target_link_libraries(${PROJECT_NAME} ${Boost_SYSTEM_LIBRARY} ${JSONCPP_LIBRARIES} ${catkin_LIBRARIES})
+target_compile_options(${PROJECT_NAME} PRIVATE -Wsuggest-override -Wconversion -Wsign-conversion)
```

Comment:
are we sue this warning override is needed?

Reply:
It actually does show many warnings for the virtual functions that are not marked as override. Should I remove it anyways?

Reply:
We could just add override, right?

Reply:
Using `override` has additional meaning. The practical reason to use it is that if you accidentally change a method's signature in the base class and not the derived class (or vice versa), then with `override` your compiler can warn you that you are not overriding anything. In other words, yes, please, add override.

Reply:
We should not add such compiler-specific options in an unconditioned fashion into the cmake files. (As far as I understand, clang uses a different option name for this.)
Although I fully agree that these warnings should be enabled, I don't like the approach enforcing them via the cmake config. Enabling or disabling warnings should be controlled from outside. For example, our Travis CI explicitly enables most warnings already:
https://github.com/ros-planning/moveit/blob/871f3a0ded8acad77c658ce70c76855bb4ff3292/.travis.yml#L22

Reply:
@ommmid, by the way, Travis still has build issues. Looks like you missed to commit `trajopt_utils/stl_to_string.hpp`:
https://travis-ci.org/ros-planning/moveit/jobs/572872763#L673-L675

Reply:
> @ommmid, by the way, Travis still has build issues. Looks like you missed to commit `trajopt_utils/stl_to_string.hpp`:
> https://travis-ci.org/ros-planning/moveit/jobs/572872763#L673-L675

The problem was "exclude" hpp file in CMakeLists. @JafarAbdi  fixed it.

Reply:
I resolved the override warnings in trajoptp but there are still warnings for override from other packages like geometric_shapes which should be dealt with in another PR